bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/discovery/ZonePlayerDiscoveryParticipant.java

```diff
@@ -82,6 +82,8 @@ public ThingUID getThingUID(RemoteDevice device) {
                     modelName = "CONNECT";
                 } else if (modelName.equals("ZP100")) {
                     modelName = "CONNECTAMP";
+                } else if (modelName.equals("One SL")) {
```

Comment:
In general we should put string  litterals first in comparisons:

```suggestion
                } else if ("One SL".equals(modelName)) {
```

Reply:
In fact, a switch/case would be even better ;)